src/audio/dsm/sof_dsm.c

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2020, Maxim Integrated
+
+#include <sof/audio/sof_dsm.h>
+#include <sof/audio/component.h>
+#include <sof/trace/trace.h>
+#include <sof/drivers/ipc.h>
+#include <sof/ut.h>
+
+#include <sof/audio/channel_map.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <ipc/channel_map.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+int sof_dsm_create(struct sof_dsm_struct_t *hsof_dsm, struct comp_dev *dev)
+{
+	if (!hsof_dsm->dsm_init) {
+		/* DSM will be initialized here. */
+		hsof_dsm->dsm_init = true;
+	} else {
+		comp_dbg(dev, "[DSM] Re-initialization.");
+	}
+	return 0;
+}
+
+void sof_dsm_ff_process(struct sof_dsm_struct_t *hsof_dsm, struct comp_dev *dev,
```

Comment:
should we maybe be abstracting this out into a backend like other components have like volume to allow for generic and SIMD implementations?

Reply:
Thanks for your comment. Let me explain more backgrounds about this commit. Maxim DSM(Dynamic Speaker Management) is speaker protection algorithm based on VI sensing feedback from smart amplifier, and it is not open source based. So, DSM algorithm is not a part of this commit. This commit only includes wrapper functions, buffering and memory allocation parts to make DSM function working in SOF framework.
I can see *_generic.c files in other component cases and it includes very generic implementation for each function but I don't have generic functions for speaker protection algorithm can be shared. Actually current commit itself is just generic code. Please let me have your thoughts for the current case.

Reply:
Thanks for the explanation. That is ok not having a generic implementation. I just want to make sure the DSM interface is properly abstracted so that if say for example maxim develops a second DSM algorithm for another chip then we can use the same DSM component interface and just switch the algorithm in the backend.

Reply:
Thanks for your comment. Let me add generic abstraction file for the speaker protection algorithm.